### ISSUE TITLE
Provide better documentation

### ISSUE THREAD

**Issue by uralets** [dosemu2/dosemu2#1602]
Specifically (but not limited to) on the topics such as:
General operation
Initialization
ALSA MIDI configuration
Maybe remove the old dosemu1 docs too - it's confusing.

Almost no docs seem to get shipped with the Ubuntu PPA packages. The README.html seems to be carried over from dosemu1.

I've decided to give dosemu2 a try playing Terra Nova (https://github.com/dosemu2/dosemu2/issues/1599 and https://github.com/dosemu2/dosemu2/issues/1600) and found myself confronted with a couple of issues and started digging for documentation and sadly found out that almost no docs are shipped with the pre-built packages neither anything was found online (in the vein of https://docs.scummvm.org/en/latest/)

For example I've tried launching DOSEMU2 without FDPP (btw how could it be disabled while it's installed? Would be nice to document this too ) bc the game mentioned above is behaving badly - and since I'm new to the fact that DOSEMU2 now has something like FDPP I've decided to experiment and use a FreeDOS installation (when I realized that FDPP sort of "replaces" it) but there's no "quick start" or anything like that and DOSEMU2 simply doesn't work out of the box without FDPP. BTW the PPA package doesn't depend on it - thus a user upon installing the package is left with a dysfunctional piece of software and virtually no clear instructions on what to do next.
DOSEMU2 is an emulator OR a virtual machine - that much is clear. And like other VMs we could try and properly install a FreeDOS inside it - from the official install media.. but should we really go through these chores for some DOS experience? You would agree it's a bit involved compared to DOSBOX.

**I do realize now having poked around that FDPP seems "the official" way to go** and I'm ofc OK with that, but you would agree the whole thing needs some high-level overview/documentation.

The Python scripts that mention FreeDOS in their names ARE getting installed but again no description how why and what for you would use them. They do seem to work (see below tho) - like download some FreeDOS-related files into .cache/ and the other one copies them into .dosemu/drive_c but still it's a mystery what exactly is going on.
Also I think I've found a bug in one of them dosemu2/install-freedos#12

As for ALSA MIDI - I have a soundcard with a real hardware MIDI (Audigy ZS 2 or something) and would like to use it for MIDI output. I do so successfully with other programs like DOSBOX and SCUMMVM (or aplaymidi for that matter).
Again, it's not clear what do you use - alsa_midi or alsa_virmidi and what's the difference between the two. Both are mentioned in the supplied dosemu.conf. But that's it. Specifically I need to point DOSEMU2 to the correct MIDI port seen in the output of aplaymidi -l. How do I do it? 
When using eg aplaymidi to play a midi file I can see

client 128: 'aplaymidi' [type=user,pid=11037]
    0 'aplaymidi       '
	Connecting To: 17:0

in the output of aconnect -l

OTOH while DOSEMU2 is working it shows
client 128: 'Client-128' [type=user,pid=11055]
    0 'Virtual RawMIDI '

Setting $_midi_driver = "alsa_midi" set in ~/.dosemurc doesn't seem to change the above output..
Grepping the sources for anything ALSA MIDI related here on github hasn't yielded anything useful... It did though made me stumble on various DOSEMU1 config bits and pieces - no idea if those are even used now?

DOSBOX-staging plays MIDI music in Terra Nova fine when the latter is configured to use General Midi (midi_num 12 in TN.CFG)
And if I clear the soundfont bank with asfxload -i the music stops playing - as it should. While in DOSEMU2 I don't hear it at all.

It seems a lot of effort is going into this project but the somewhat haphazard documentation IMHO hampers it.

I sure would like to play some DOS games with significantly less strain on the CPU - something that DOSBOX(-staging) authors has never seen as a priority (or even a possibility) AFAIK.

**Comment by stsp** [comment]
> Maybe remove the old dosemu1 docs too - it's confusing.

Yes, that should be done.

> The Python scripts that mention FreeDOS in their names ARE getting installed but again no description how why and what for 

But there is:
https://github.com/dosemu2/dosemu2/wiki/installing-various-DOSes
Our wiki is still very poorly structured and
has a notably bad articles too, but here it is.

> It seems a lot of effort is going into this project but the somewhat haphazard documentation IMHO hampers it.

People who seek the documentation, are
just looking for problems. dosemu2 should
work out of the box. Which is why the documentation
is not within, but in a wiki. The user should start
w/o any docs and just _try_ things out. If he doesn't
try and instead is looking for a documentation, he
already lost the battle. :) But as soon as he tried
and made sure the basic things work, then he may
start wandering around wiki for some advanced topics.

> I sure would like to play some DOS games with significantly less strain on the CPU

Not sure if you can have a good power saving
with Terra Nova game on dosemu2 any soon,
sorry about that. We can try to make it run faster
(though I don't see sluggishness that you mention)
but power saving is not a priority right now.

> DOSBOX-staging plays MIDI music in Terra Nova fine when the latter is configured to use General Midi

Same with dosemu2 - without any configuration
or documentation. It just works, I have checked.

**Comment by stsp** [comment]
I mean, just how much documentation do
you read before starting firefox? I guess none,
it just works. Only when it doesn't, you go
to google. But in most cases you end up in
KB or bugzilla, not in a documentation, anyway.

**Comment by stsp** [comment]
> For example I've tried launching DOSEMU2 without FDPP (btw how could it be disabled while it's installed?

If there is some other DOS on drive
C, then it boots. If there is some other
command.com on drive C, then its used.
fdpp or comcom32 never come ahead
of anything you copied to drive C.
The aforementioned wiki page mentions
that fact.
Any documentation that you see missing,
can be written by yourself on our wiki,
with a big thank you for your help. :)
